### PR TITLE
Address issues from first rev.

### PR DESCRIPTION
**Pull request by Avielyo10**

### CODE REVIEW

File: gqlex.g4 [hunk 1]
```diff
@@ -40,19 +76,19 @@ pathExpression
     ;
 
 rootPathExpression
-    : (ALL | NODE) (pathElement (NODE pathElement)*)+
+    : (DOUBLE_SLASH | SLASH) (pathElement (SLASH pathElement)*)?
```
Comment (yaronf) [inline gqlex.g4:79]: @yaronTTkarni Do we allow `//` (no range, no path elements) as a valid expression?

File: gqlex.g4 [hunk 2]
```diff
@@ -71,13 +107,22 @@ conditionLogicalOperator
     ;
 
 conditionExpression
-    : lValue conditionOperator rValue (conditionLogicalOperator conditionExpression)?
+    : nameValue conditionOperator (nameValue | intValue)
+    | conditionExpression conditionLogicalOperator conditionExpression
```
Comment (yaronf) [inline gqlex.g4:112]: Even if we ignore operator precedence for now, we should support parentheses in expressions.

File: gqlex.g4 [hunk 3]
```diff
@@ -40,19 +76,19 @@ pathExpression
...
     ;
 
 elementSelectionExpression
-    : lValue EQUALS rValue
+    : nameValue EQUALS nameValue
     ;
 
 pathElement
-    : lValue (LBRACK elementSelectionExpression RBRACK)? (LPAREN conditionExpression RPAREN)?
+    : nameValue (LBRACK elementSelectionExpression RBRACK)? (LPAREN conditionExpression RPAREN)?
```
Comment (yaronf) [inline gqlex.g4:91]: I suggest to rename conditionOperator to comparisonOperator.

**Review by yaronf — CHANGES_REQUESTED** [review]

File: Makefile
Comment (yaronf) [inline Makefile:1]: Please also expose the -package flag of antlr4, it is necessary so the generated code can be used without change.

File: Makefile [hunk 4]
```diff
@@ -14,3 +14,8 @@ clean:
 install:
 	@echo "Installing dependencies..."
 	@pip install antlr4-tools
+
+test_cases:
+	@echo "Generating test cases..."
+	@grammarinator-process gqlex.g4 -o tests/fuzzer/ --no-actions
```
Comment (yaronf) [inline Makefile]: Based on the error message, I think you want `-r document` here, too.

File: gqlex.g4 (hunk 1 above)
Comment (yaronTTkarni) [reply gqlex.g4:79]: yes, most simple query

File: gqlex.g4 [hunk 5]
```diff
@@ -71,13 +107,22 @@ conditionLogicalOperator
     ;
```
Comment (yaronTTkarni) [inline gqlex.g4:107]: @Avielyo10 assume the condition is empty now

File: Makefile
Comment (Avielyo10) [reply Makefile:1]: Done

File: Makefile (hunk 4 above)
Comment (Avielyo10) [reply Makefile]: What error message did you get?
I added venv with its deps. so now `make` should run the same for any env

File: gqlex.g4 (hunk 3 above)
Comment (Avielyo10) [reply gqlex.g4:91]: Done

File: Makefile (hunk 4 above)
Comment (yaronf) [reply Makefile]: The farthest rule from 'NEGATIVE_SIGN' is 'NEGATIVE_SIGN' (0 steps).

42 rule(s) unreachable from 'NEGATIVE_SIGN': 'EOF', 'NONZERO_DIGIT', 'DIGIT', 'LETTER', 'NAME_START', 'NAME_CONTINUE', 'SLASH', 'DOUBLE_SLASH', 'DOTS', 'RANGE_OPEN', 'RANGE_CLOSE', 'COLON', 'EQUALS', 'LBRACK', 'RBRACK', 'LPAREN', 'RPAREN', 'COND_EQ', 'COND_NE', 'COND_GT', 'COND_GE', 'COND_LT', 'COND_LE', 'COND_AND', 'COND_OR', 'WS', 'POSITIVE_INT', 'NEGATIVE_INT', 'NAME', 'document', 'pathExpression', 'rootPathExpression', 'rangePathExpression', 'elementSelectionExpression', 'pathElement', 'conditionOperator', 'conditionLogicalOperator', 'conditionExpression', 'nameValue', 'intValue', 'rangeEndIntValue', 'rangeStartIntValue'

Comment (yaronf) [reply Makefile]: You have `-r document` for the second commend but not for the first one. I think you should have it in both.

Comment (Avielyo10) [reply Makefile]: done

**Pull request merged**